python/ray/tests/test_cancel.py

```diff
@@ -125,12 +125,19 @@ def test_defer_sigint():
 
 
 def test_defer_sigint_monkey_patch():
-    # Tests that setting a SIGINT signal handler within a DeferSigint context is not
-    # allowed.
+    # Tests that setting a SIGINT signal handler within a DeferSigint
+    # context will defer setting it until the context exits.
     orig_sigint_handler = signal.getsignal(signal.SIGINT)
-    with pytest.raises(ValueError):
-        with DeferSigint():
-            signal.signal(signal.SIGINT, orig_sigint_handler)
+    new_sigint_handler = lambda signum, frame: None  # noqa: E731
+
+    with DeferSigint():
+        assert signal.signal(signal.SIGINT, new_sigint_handler) is orig_sigint_handler
```

Comment:
I think we should not test the implementation detail but the behavior of the code:

```
with DeferSigint():
    signal.signal(signal.SIGINT, my_handler)
    os.kill(myself, SIGINT)
    assert_my_handler_not_called()
assert_my_handler_called()
```

Reply:
Done